bittensor/_dataset/dataset_impl.py

```diff
@@ -1,678 +1,750 @@
-            logger.error(f"Failed to get from IPFS {file_meta['Name']} {E}")
-            return None
+        self.datasets = self.datasets[:self.max_datasets]
+        self.no_tokenizer = no_tokenizer
+        self.buffer_calls_per_update = buffer_calls_per_update
+        self.sample_cat_tasks = []
+        self.sample_buffer = []
 
-        return response
+        # Tokenizer 
+        self.tokenizer =  bittensor.tokenizer()
+        self.pad_token = self.tokenizer.pad_token
+        self.pad_token_idx = self.tokenizer(self.pad_token)['input_ids'][0]
 
-    def __len__(self):
-        """ Returns length of the dataset that the dataset is processing
-        """
+        self.data_dir =  data_dir
+        self.save_dataset = save_dataset
+        self.load_dataset = load_dataset
 
-    def __getitem__(self, idx):
-        """ Returns the next batch from the dataset.
-        """
+        # set the buffer
+        self.set_buffer(buffer_size=buffer_size)
 
-class GenesisTextDataset( Dataset ):
-    """ One kind of dataset that caters for the data from ipfs 
-    """
-    def __init__(
-        self,
-        block_size,
-        batch_size,
-        num_workers,
-        dataset_name,
-        data_dir,
-        save_dataset,
-        max_datasets,
-        no_tokenizer, 
-        num_batches,
-        max_directories
-    ):
-        super().__init__()
-        self.block_size = block_size
-        self.batch_size = batch_size
-        self.num_workers = num_workers
-        self.tokenizer = bittensor.tokenizer( version = bittensor.__version__ )
-        self.dataset_name = dataset_name
-        self.data_dir = data_dir
-        self.save_dataset = save_dataset
-        self.datafile_size_bound = 262158
-        self.max_datasets = max_datasets
-        self.__infinite_dataset_iterator = None
-        self.no_tokenizer = no_tokenizer
-        self.IPFS_fails = 0
-        self.backup_dataset_cap_size = 5e7 # set 50MB limit per folder
-        self.IPFS_fails_max = 10
+        # TODO: currently the number of batches is inert as this loop runs forever
         self.num_batches = num_batches
-        self.max_directories = max_directories
+        self.sample_count = 0
+        self.batch_count = 0
 
-        # Retrieve a random slice of the genesis dataset
-        self.data = []
-        self.data_reserved = []
+        # Build the text corpus by fetching the hashes of the textfiles (Current Heirarchy)
+        self.construct_text_corpus(datasets=self.datasets, load=self.load_dataset, save=self.save_dataset)
 
-        # Used to refresh corpus if we've exhausted the whole dataset
-        self.refresh_corpus = True
+    def construct_text_corpus(self, datasets:Optional[List[str]] = None, save:bool = False, load:bool = False) -> None :
+        """ Building all of the datasets specified by getting each of their 
+            text hashes from IPFS or local
+        Args:
+            datasets (List[str]):
+                Axon to serve.s
```

Comment:
This is the type of mistake that we cannot have on master. Stopping my review here.